lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -293,11 +295,25 @@ public LockRefreshToken lockAnonymously(LockRequest request) throws InterruptedE
     @Override
     public LockRefreshToken lockWithClient(String client, LockRequest request)
             throws InterruptedException {
+        Preconditions.checkArgument(request.getLockGroupBehavior() == LockGroupBehavior.LOCK_ALL_OR_NONE,
+                "lockWithClient() only supports LockGroupBehavior.LOCK_ALL_OR_NONE. Consider using tryLockWithClient().");
```

Comment:
Another way to do this is to remove LockGroupBehavior from LockRequest and stick it as an argument on the tryLock methods.